Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -163,17 +162,21 @@ static bool hasCNAMEAndCookieTransformSPI(NSURLSessionDataTask* task)
 
 void NetworkDataTaskCocoa::updateFirstPartyInfoForSession(const URL& requestURL)
 {
-    if (!hasCNAMEAndCookieTransformSPI(m_task.get()) || !networkSession() || !networkSession()->networkStorageSession() || !networkSession()->networkStorageSession()->resourceLoadStatisticsEnabled() || requestURL.host().isEmpty())
+    auto* session = networkSession();
+    if (!session || !session->networkStorageSession() || !session->networkStorageSession()->resourceLoadStatisticsEnabled() || requestURL.host().isEmpty())
```

Comment:
I wonder if the repeated use of this check for an active session and ITP should be broken out into a convenience function?

Reply:
Yes! (I was planning to do that as a part of the next patch, though we can pull that into this instead). Will pull this out into a separate helper method, called `shouldApplyCookiePolicyForThirdPartyCNAMECloaking()`.